CNCF TAG Observability WG: charter for observe-k8s

halcyondude opened the pull request.

kenfinnigan commented on charter.md:
```diff
@@ -0,0 +1,355 @@
+* This document has images: check for >>>>>  gd2md-html alert:  inline image link in generated source and store images to your server. NOTE: Images in exported zip file from Google Docs may not appear in  the same order as they do in your doc. Please check the images!
+----->
+
+<p style="color: red; font-weight: bold">>>>>>  gd2md-html alert:  ERRORs: 0; WARNINGs: 0; ALERTS: 2.</p>
```
Should we remove warnings/alerts from the generation process?

halcyondude replied:
Yes indeed  :)

kenfinnigan commented on charter.md:
```diff
@@ -0,0 +1,355 @@
+
+This working group aims to provide:
+
+- a set suite of realistic sample applications, which employ observability tools to demonstrate how practitioners can gain insight and understanding.
```
```suggestion
- a set of realistic sample applications, which employ observability tools to demonstrate how practitioners can gain insight and understanding.
```

kenfinnigan commented on charter.md:
```diff
@@ -0,0 +1,355 @@
+
+    (source: https://github.com/cncf/toc/blob/main/tags/cncf-tags.md#responsibilities--empowerment-of-tags)
+
+1. Charter / Description of WG
```
Looks like the nested lists got messed up in the generation.

halcyondude replied:
aye, will fix

mhausenblas commented on charter.md:
```diff
@@ -0,0 +1,336 @@
+This working group aims to provide:
+
+- a set suite of realistic sample applications, which employ observability tools to demonstrate how practitioners can gain insight and understanding.
+- actionable, best-practice guidance on how to apply observability to real-life workloads. Observed workloads will be CNCF Projects.
```
Maybe use "good practices" rather than "best practices"? Our space is very new and I doubt that any of us can claim to know what is best in absolute terms.

halcyondude replied:
done

mhausenblas commented on charter.md:
```diff
@@ -0,0 +1,336 @@
+# observe-k8s working document
+
+**Table of Contents**
```
I think the ToC is way too detailed. Let's prune it so that the important bits are not buried.

halcyondude replied:
agreed, for now nuking it entirely.

mhausenblas commented on charter.md:
```diff
@@ -0,0 +1,336 @@
+- Matt Young ([halcyondude@gmail.com](mailto:halcyondude@gmail.com))
+- Ivan Santos ([pragmaticivan@gmail.com](mailto:pragmaticivan@gmail.com), EverQuote)
+- Ken Finnigan ([ken@kenfinnigan.me](mailto:ken@kenfinnigan.me), Workday)
+- Michael Hausenblas ([mh9@o11y.engineering](mailto:mh9@o11y.engineering), AWS)
```
Please use my work address, `hausenbl@amazon.com`

halcyondude replied:
done

mhausenblas commented on charter.md:
```diff
@@ -0,0 +1,336 @@
+- Daniel Khan ([daniel@khan.io](mailto:daniel@khan.io), Elastic)
+- Severin Neumann ([severin.neumann@altmuehlnet.de](mailto:severin.neumann@altmuehlnet.de), Cisco)
+
+## Discussion(s) thus far:
```
Either expand on this section or remove. As it stands, I doubt the bullet list is helpful for the reader.

halcyondude replied:
agreed and removed.  Left a few concrete ddetails from gitops wg and cncf.

I think this document should be cleaved into a charter that is only mission statement + charter, and the rest can seed a design TLD.

mhausenblas commented on charter.md:
```diff
@@ -0,0 +1,336 @@
+
+- Charter / Description of WG (_This document once it's merged_)
+- Time Box:
+  - 2 quarters (Q1-Q2 2022)
```
Given timeline, approval etc. should we start with Q2?

halcyondude replied:
that's a great point - I'm not sure how long it'll take to be formally approved.

Setting the terminus --> Q3 makes this ready for a GA/reveal for fall conf season.

mhausenblas commented on charter.md:
```diff
@@ -0,0 +1,336 @@
+- Engage with SIG Instrumentation re: gaps and opportunities for signals from k8s
+- Collect and Curate templates, recipes, examples, demos highlighting the observation of cloud native systems and workloads (CNCF Projects).
+
+## Personas
```
I agree that this section is important but as it stands it's not very useful. Let's either cut it down to "Gibbs is on it" or flesh it out.

halcyondude replied:
done

mhausenblas commented on charter.md:
```diff
@@ -0,0 +1,336 @@
+
+_Proposed: Brainstorming - this is not authoritative!_
+
+### Architecture
```
Not sure what this is, remove?

halcyondude replied:
done

mhausenblas commented on charter.md:
```diff
@@ -0,0 +1,336 @@
+
+TODO: 10k' Diagram
+
+### GitOps for $allTheThings
```
Remove, not relevant for now.

halcyondude replied:
done

mhausenblas commented on charter.md:
```diff
@@ -0,0 +1,336 @@
+
+todo image
+
+### Potential example workloads
```
That section should be more prominent, there was consensus on the potential workloads candidate and this is kind of core.

halcyondude replied:
agreed and done

mhausenblas reviewed (changes requested):
Thanks for kicking this off! Left comments inline.

halcyondude commented:
I've reorganized the document per feedback, fixed some of the lingering table/list formatting issues, and addressed feedback given (thank you!) thus far.

kenfinnigan commented on charter.md:
```diff
@@ -0,0 +1,210 @@
+- Time Box:
+  - 3 quarters (Q1-Q3 2022)
+- Deliverables:
+  - github.com/observe-k8s: An umbrella organization for repos that realize the goals of this org
```
I wasn't sure if "org" should be "working group"? It felt self circular otherwise

mhausenblas replied:
I interpreted "org" as being short for "this GitHub organization" (`observe-k8s`).

kenfinnigan replied:
That's how I read it, but it didn't make sense to me.

Are the goals of the working group the same goals as the GH org, or are they different? Or is it to allow for the working group to foster the creation of the GH org to live its own life? So initially the goals are aligned but over time, perhaps once the working group is disbanded, the GH org goals will move on and evolve?

dofinn commented on charter.md:
```diff
@@ -0,0 +1,210 @@
+This working group aims to provide:
+
+- a suite of realistic sample applications, which employ observability tools & techniques to demonstrate how practitioners can gain insight and understanding.
+- actionable, good practice & peer reviewed guidance on how to enable observability for real-life workloads. Observed workloads will be CNCF Projects.
```
> Observed workloads will be CNCF Projects
what is ment by this exactly?

mhausenblas replied:
That we try to focus for the system under observation (SuO) on CNCF projects. For example, a Kubernetes workload would be in scope, some proprietary provider API for running the workloads less so.

dofinn replied:
Understood :) Thanks @mhausenblas - id be interested in pushing the envelope her from an SRE persona where running a single kube cluster VS a fleet changes observavility requirements (if this is something observe-k8s would be interested in)

halcyondude commented:
addressed comments, merging
